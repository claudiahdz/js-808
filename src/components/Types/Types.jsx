import React from 'react'
import PropTypes from 'prop-types'

const Types = ({ changeVolume, volumes }) => {
  const types = ['Kick', 'Snare', 'Open Hat', 'Closed Hat']

  return (
    <div className="types">
      { types.map((type, i) => { return (
          <div className="type" key={i}>
            {type}
            <input type="range" value={volumes[i]} onChange={(e) => changeVolume(i, e.target.value)} /> 
            <div>Volume</div>
          </div>
        )}) 
      }
    </div>
  )
}

Types.propTypes = {
  changeVolume: PropTypes.func.isRequired,
}


export default Types
