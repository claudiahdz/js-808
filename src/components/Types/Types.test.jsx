import React from 'react'
import { shallow } from 'enzyme'
import Types from './Types'

describe('Types', () => {
  const defaultProps = {
    volumes: [100,100,100,100],
    changeVolume: () => {}
  }

  it('renders the component correctly', () => {
    const types = shallow(<Types {...defaultProps} />)
    expect(types).toMatchSnapshot()
  })

  it('changes volume', () => {
    const props = {
      ...defaultProps,
      changeVolume: jest.fn()
    }
    const e = { target: {value: 60} }
    const types = shallow(<Types {...props} />)
  
    const input = types.find('input')
    input.first().simulate('change', e)

    expect(props.changeVolume).toHaveBeenCalledWith(0, e.target.value)
  })
})

