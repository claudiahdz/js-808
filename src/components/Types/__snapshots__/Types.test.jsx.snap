// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Types renders the component correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="types"
>
    <div
        className="type"
    >
        Kick
        <input
            defaultValue="100"
            onChange={[Function]}
            type="range"
        />
    </div>
    <div
        className="type"
    >
        Snare
        <input
            defaultValue="100"
            onChange={[Function]}
            type="range"
        />
    </div>
    <div
        className="type"
    >
        Open Hat
        <input
            defaultValue="100"
            onChange={[Function]}
            type="range"
        />
    </div>
    <div
        className="type"
    >
        Closed Hat
        <input
            defaultValue="100"
            onChange={[Function]}
            type="range"
        />
    </div>
</div>,
  "nodes": Array [
    <div
      className="types"
>
      <div
            className="type"
      >
            Kick
            <input
                  defaultValue="100"
                  onChange={[Function]}
                  type="range"
            />
      </div>
      <div
            className="type"
      >
            Snare
            <input
                  defaultValue="100"
                  onChange={[Function]}
                  type="range"
            />
      </div>
      <div
            className="type"
      >
            Open Hat
            <input
                  defaultValue="100"
                  onChange={[Function]}
                  type="range"
            />
      </div>
      <div
            className="type"
      >
            Closed Hat
            <input
                  defaultValue="100"
                  onChange={[Function]}
                  type="range"
            />
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Types
        changeVolume={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "changeVolume": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="types"
>
          <div
                    className="type"
          >
                    Kick
                    <input
                              defaultValue="100"
                              onChange={[Function]}
                              type="range"
                    />
          </div>
          <div
                    className="type"
          >
                    Snare
                    <input
                              defaultValue="100"
                              onChange={[Function]}
                              type="range"
                    />
          </div>
          <div
                    className="type"
          >
                    Open Hat
                    <input
                              defaultValue="100"
                              onChange={[Function]}
                              type="range"
                    />
          </div>
          <div
                    className="type"
          >
                    Closed Hat
                    <input
                              defaultValue="100"
                              onChange={[Function]}
                              type="range"
                    />
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="types"
>
          <div
                    className="type"
          >
                    Kick
                    <input
                              defaultValue="100"
                              onChange={[Function]}
                              type="range"
                    />
          </div>
          <div
                    className="type"
          >
                    Snare
                    <input
                              defaultValue="100"
                              onChange={[Function]}
                              type="range"
                    />
          </div>
          <div
                    className="type"
          >
                    Open Hat
                    <input
                              defaultValue="100"
                              onChange={[Function]}
                              type="range"
                    />
          </div>
          <div
                    className="type"
          >
                    Closed Hat
                    <input
                              defaultValue="100"
                              onChange={[Function]}
                              type="range"
                    />
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Types
    changeVolume={[Function]}
/>,
}
`;
